src/SMU-Keithley_6430/main.py

```
@@ -191,46 +328,76 @@ def reach(self) -> None:
+            self.time_stamps = values[3::5]
 
-    def call(self) -> list[float]:
+        else:
+            timeout = max(self.speed * self.averages * 3, 3)  # seconds, allow 3x expected time plus a fixed 3 s margin
+            time_start = time.monotonic()
+
+            while not self.is_run_stopped() and (time.monotonic() - time_start) < timeout:
+                try:
+                    results = self.port.read()
+                    break
+                except Exception as e:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)